product_unique_serial/tests/test_common.py

```diff
@@ -0,0 +1,123 @@
+            # Getting default data through product_uom_qty onchange
+            default_qty_data = self.stock_move_obj.onchange_quantity(
+                move_n.get('product_id'),
+                qty,
+                default_product_data.get('product_uom'),
+                default_product_data.get('product_uos'))
+            default_qty_data['value'].update({'product_uom_qty': qty})
+            move_n.update(default_qty_data.get('value'))
+            # Getting the new stock.move id
+            move_created = self.stock_move_obj.with_context(
+                default_picking_type_id=picking_type.id).create(move_n)
+            stock_move_ids.append(move_created.id)
+        # Creating picking
+        picking_data_copy.update({
+            'move_lines': [(6, 0, stock_move_ids)],
+            # TODO: Uncomment when fix current context bug
+            # 'move_lines': [(0, 0, moves_data_copy)],
+            # and remove move_created line
+            'picking_type_id': picking_type.id})
+        return self.stock_picking_obj.create(picking_data_copy)
+
+    def transfer_picking(self, picking_instance, serial_number=None):
+        """ Creates a wizard to transfer the picking given like parameter """
+        # Marking the picking as Todo
+        picking_instance.action_confirm()
+        if picking_instance.state == 'confirmed':
+            # Checking availability
+            picking_instance.action_assign()
+        if picking_instance.state == 'assigned':
+            # Transfering picking
+            transfer_details = picking_instance.do_enter_transfer_details()
+            wizard_for_transfer = self.env[transfer_details.get('res_model')].\
+                browse(transfer_details.get('res_id'))
+            if serial_number:
+                if len(serial_number) == 1:
+                    for transfer_item in wizard_for_transfer.item_ids:
+                        transfer_item.lot_id = serial_number[0].id
+                else:
+                    # Case for the A, B & C serial numbers: That are 3 like
+                    # value for the quantity field in the wizard, so we should
+                    # split this record 2 times.
+                    # Splitting the record and executing the transfer
+                    wizard_split = wizard_for_transfer.item_ids[0].\
+                        split_quantities()
+                    wizard_for_transfer = self.env['stock.transfer_details'].\
+                        browse(wizard_split.get('res_id'))
+                    wizard_split = wizard_for_transfer.item_ids[0].\
+                        split_quantities()
+                    wizard_for_transfer = self.env['stock.transfer_details'].\
+                        browse(wizard_split.get('res_id'))
+                    index = 0
+                    for transfer_item in wizard_for_transfer.item_ids:
+                        transfer_item.lot_id = serial_number[index].id
+                        index += 1
```

Comment:
No men.

Either the code is test or is program..

If Program then need test if test is because it assert the output of a program no there are such thing as Limbo in that statement.

Even if it is common code for tests the if is not correct, then you must mode those algorithms to your models or redegising correctly it.

@suniagajose and @luistorresm know my position about that.